iamy/iam.go

```diff
@@ -25,12 +25,19 @@ func (c *iamClient) getPolicyDescription(arn string) (string, error) {
 	return "", err
 }
 
-func (c *iamClient) getRoleDescription(name string) (string, error) {
+func (c *iamClient) getRole(name string) (string, int, error) {
 	resp, err := c.GetRole(&iam.GetRoleInput{RoleName: &name})
-	if err == nil && resp.Role != nil && resp.Role.Description != nil {
-		return *resp.Role.Description, nil
+	var sessionDuration int64
+	var description string
+	// 3600 is the default, so let's ignore it
+	if resp.Role.MaxSessionDuration != nil && *resp.Role.MaxSessionDuration != 3600 {
```

Comment:
Given we are doing some validation, is it worth chucking a wobbly if the value supplied is larger than permitted?  I guess we'd check the YAML not here, but you get what I mean.

Reply:
I'd rather let the API do that validation.

Reply:
same is true of the default,  but it's definitely not a hill to even be injured on :grin: